### PR TITLE
Create area_calc

### PR DESCRIPTION
**Pull request by s3qn**

### CODE REVIEW

**Comment by s3qn** [comment]
@akholodny

**Comment by akholodny** [comment]
Please rename the file (add ".py")

File: area_calc.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+shape = input('Enter shape to calculate, must be "SQUARE", "TRIANGLE", "RECTANGLE", "TRAPEZOID":')
+
+class Calc():
+
+    base = int(input('Enter base:'))
+    height = int(input('Enter height:'))
```
Comment (akholodny) [inline area_calc.py]: Move to the constructor of base class

File: area_calc.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    if shape == 'TRAPEZOID' : base2 = int(input('Enter second base size:'))
```
Comment (akholodny) [inline area_calc.py]: move the logic to the constructor Trapezoid (override the constructor of base class in child class Trapezoid)

File: area_calc.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    def __init__(self, height, base):
+        self.height = height
+        self.base = base
+
+class Square(Calc):
+
+    def cal():
+        return print('Calculated square area:', Calc.base * Calc.height)
```
Comment (akholodny) [inline area_calc.py]: return value only (without print)

File: area_calc.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    height = int(input('Enter height:'))
+    if shape == 'TRAPEZOID' : base2 = int(input('Enter second base size:'))
+
+    def __init__(self, height, base):
+        self.height = height
+        self.base = base
+
+class Square(Calc):
+
+    def cal():
```
Comment (akholodny) [inline area_calc.py]: rename the method `cal` to `getArea`

File: area_calc.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        return print('Calculated square area:', Calc.base * Calc.height)
+
+class Triangle(Calc):
+
+    def cal():
+        return print('Calculated triangle area:', Calc.base * Calc.height / 2)
```
Comment (akholodny) [inline area_calc.py]: Don't use ` Calc.base` ,  Calc.height, etc  (use self.base, self.height, etc)

File: area_calc.py [hunk 6]
```diff
@@ -2,43 +2,50 @@
...
+    def getArea(self):
+        return self.base * self.height
 
 
 class Trapezoid(Calc):
 
-    def cal(self):
-        return print('Calculated trapezoid area:', (Calc.base + Calc.base2) * Calc.height / 2)
+     def __init__(self, height, base, base2):
+         super().__init__(height, base)
```
Comment (akholodny) [inline area_calc.py]: There is not the constructor with two input parameters in base class

File: area_calc.py [hunk 7]
```diff
@@ -2,43 +2,50 @@
...
-        return print('Calculated rectangle area:', Calc.base * Calc.height)
+    def getArea(self):
+        return self.base * self.height
 
 
 class Trapezoid(Calc):
 
-    def cal(self):
-        return print('Calculated trapezoid area:', (Calc.base + Calc.base2) * Calc.height / 2)
+     def __init__(self, height, base, base2):
```
Comment (akholodny) [inline area_calc.py]: override constructor of base class (pay attention to  prototype of  constructor )

File: area_calc.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+        print ("Trapezoid Area is", trapezoid.getArea(), 'Type "DONE" when finished.')
+        results.append(trapezoid.getArea())
+        continue
+    elif shape == 'DONE': break
+
+    else:
+        print('Unrecognized shape, try again.')
+        continue
+
+print('Process done! \nMaximum element is:', max(results),'\nMinimum element is:',min(results))
```
Comment (akholodny) [inline area_calc.py]: According to the task: 

> 2) find object with minimum area
> 3) find object with maximum are"

Found the area only (but not type)

File: area_calc.py [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+        print ("Trapezoid Area is", trapezoid.getArea(), 'Type "DONE" when finished.')
+        area = (trapezoid.getArea(),'Trapezoid')
+        results.append(area)
+        continue
+    elif shape == 'DONE': break
+
+    else:
+        print('Unrecognized shape, try again.')
+        continue
+
```
Comment (akholodny) [inline area_calc.py:68]: move to `results.append(area)` (from line 43, 49, 55 and 61)

File: area_calc.py [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+results = []
+while True:
+    shape = input('Enter shape to calculate, must be "SQUARE", "TRIANGLE", "RECTANGLE" or "TRAPEZOID":')
+
+    if shape == 'SQUARE':
+        square = Square()
+        print ("Square Area is" , square.getArea(), 'Type "DONE" when finished.')
+        area = (square.getArea(),'Square')
+        results.append(area)
+        continue
```
Comment (akholodny) [inline area_calc.py]: remove `continue` (and from lines 50, 56, 62) (isn't needed)

File: area_calc.py (hunk 8 above)
Comment (akholodny) [reply area_calc.py]: Fix error if `results` is empty

File: area_calc.py [hunk 11]
```diff
@@ -0,0 +1,69 @@
+class Calc():
```
Comment (akholodny) [inline area_calc.py]: Remove `()` (isn't needed , because python 3)

File: area_calc.py [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+        rectangle = Rectangle()
+        print ("Rectangle Area is", rectangle.getArea(), 'Type "DONE" when finished.')
+        area = (rectangle.getArea(),'Rectangle')
+
+    elif shape == 'TRAPEZOID':
+        trapezoid = Trapezoid()
+        print ("Trapezoid Area is", trapezoid.getArea(), 'Type "DONE" when finished.')
+        area = (trapezoid.getArea(),'Trapezoid')
+
+    elif shape == 'DONE': break
```
Comment (akholodny) [inline area_calc.py]: Fix code style (move `break` to new line)

File: area_calc.py [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+
+    else:
+        print('Unrecognized shape, try again.')
+        continue
+    results.append(area)
+try:
+    print('Process done! \nMaximum element is:', max(results),'\nMinimum element is:',min(results))
+
+except:
+    print('Error: No result found')
```
Comment (akholodny) [inline area_calc.py]: add exception message

**Review by akholodny — APPROVED** [review]
Approved. can be merged.

**Pull request merged**